app/models/manageiq/providers/kubevirt/infra_manager/metrics_capture.rb

```diff
@@ -0,0 +1,141 @@
+class ManageIQ::Providers::Kubevirt::InfraManager::MetricsCapture < ManageIQ::Providers::InfraManager::MetricsCapture
```

Comment:
@nasark how much of this is copy&paste from `ManageIQ::Providers::Kubernetes::ContainerManager::MetricsCapture` ?

I wonder if something like a `PrometheusMetricsCaptureMixin` might help reduce the overlap?

Reply:
Yeah there's a lot of overlap. I wasn't sure if subclassing the k8s metrics capture was the right move since its `ContainerManager::MetricsCapture` but you can let me know. If you mean `PrometheusClientMixin` and `CaptureContextMixin`, I've overridden what was necessary in `prometheus_capture_context.rb` https://github.com/ManageIQ/manageiq-providers-kubevirt/pull/260/files#diff-20a87c8350c78e56eae548e8c7dbb3558e278aaed713954c3b711826c45c18f0

Reply:
Yeah don't think we should subclass it since like you said that would be under the ContainerManager class tree not the InfraManager one.  I was thinking a mixin in the kubernetes provider could take care of this but I'm okay with that being a follow-up too.